### PR TITLE
feat: add a RecordResource to log the received record from tis stage

### PR DESCRIPTION
**Pull request by hihilary**
No more logic is needed this sprint, just created a api to make sure the record sent by aws lambda is received by this service.

TISNEW-4306

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -42,6 +42,10 @@ dependencies {
 
   // Sentry reporting
   compile "io.sentry:sentry-spring:1.7.30"
+
+  // test
+  testImplementation('org.junit.jupiter:junit-jupiter-engine:5.2.0')
```
Comment (Judge40) [inline build.gradle:47]: These should not be needed, the test starter is included at line 30 already.

File: src/main/java/uk/nhs/hee/tis/trainee/sync/api/RecordResource.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+public class RecordResource {
+
+  private static final Logger log = LoggerFactory.getLogger(RecordResource.class);
+
+  public RecordResource() {}
+
+  @PostMapping("/record")
+  public ResponseEntity<RecordDto> receiveRecord(
+      @RequestBody RecordDto recordDto
+  ) {
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/sync/api/RecordResource.java:23]: Short enough to be a single line.

File: src/test/java/uk/nhs/hee/tis/trainee/sync/api/RecordResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import uk.nhs.hee.tis.trainee.sync.dto.RecordDto;
+
+@ExtendWith(SpringExtension.class)
+@WebMvcTest(controllers = RecordResource.class)
+public class RecordResourceTest {
+
+  private static final String DEFFAULT_RECORD_ID = "DEFFAULT_RECORD_ID";
+  private static final String DEFFAULT_SCHEMA_NAME = "DEFFAULT_SCHEMA_NAME";
```
Comment (Judge40) [inline src/test/java/uk/nhs/hee/tis/trainee/sync/api/RecordResourceTest.java:24]: Typos in "DEFAULT"

File: src/main/java/uk/nhs/hee/tis/trainee/sync/api/RecordResource.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.nhs.hee.tis.trainee.sync.dto.RecordDto;
+
+@RestController
+@RequestMapping("/api")
+public class RecordResource {
+
+  private static final Logger log = LoggerFactory.getLogger(RecordResource.class);
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/sync/api/RecordResource.java:16]: Logger is a constant so should be all upper case "LOG"

File: build.gradle (hunk 1 above)
Comment (hihilary) [reply build.gradle]: Removed, cheers.

File: src/main/java/uk/nhs/hee/tis/trainee/sync/api/RecordResource.java (hunk 4 above)
Comment (hihilary) [reply src/main/java/uk/nhs/hee/tis/trainee/sync/api/RecordResource.java]: fixed, thanks!

File: build.gradle [hunk 5]
```diff
@@ -42,6 +42,9 @@ dependencies {
 
   // Sentry reporting
   compile "io.sentry:sentry-spring:1.7.30"
+
+  // test
+  testImplementation('org.mockito:mockito-junit-jupiter:2.23.0')
```
Comment (Judge40) [inline build.gradle:47]: Mockito is also included in the spring test starter, any reason we can't use that and not declare the dependency again?

Comment (hihilary) [reply build.gradle:47]: I thought it wasn't included, will remove it as well. :)

**Comment by hihilary** [comment]
To fix the vulnerability reported by Sonar, I added an entity and a mapper for the Dto.

**Pull request merged**